### PR TITLE
add Google Season of Docs project proposal page

### PR DESCRIPTION
**Pull request by xmulligan**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

```release-note
added Google Season of Docs Project proposal page
```

### CODE REVIEW

File: Documentation/contributing/development/season_of_docs.rst [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+Cilium is interested in applying to the 2022 Google Season of Docs to improve the documentation experience for our users and contributors.
+Below is our project ideas, if you have any questions, please reach out to gsod@cilium.io.
+
+Reorganize Cilium Features and Getting Started 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+About Our Organization
+~~~~~~~~~~~~~~~~~~~~~~
+
+Cilium (current version 1.12, first released in 2017) is an Apache 2.0-licensed project for providing,
```
Comment (joestringer) [inline Documentation/contributing/development/season_of_docs.rst]: nit: For what it's worth, 1.12 is the current development version, 1.11 is the latest GA release at this time. We plan to release 1.12 during the program, possibly also 1.13 but we have not yet planned out anything for that version.

File: Documentation/contributing/development/season_of_docs.rst [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+~~~~~~~~~~~~~~~~~
+
+Cilium currently receives issues covered in the documentation in both the Slack and through Github issues. It also receives few new 
+contributors to the documentation. We believe that this restructuring will result in fewer questions and issues already covered in 
+the documentation and more contributors to the documentation since they will know where to add it.
+
+We will track three metrics (number of questions covered in documentation in Slack, number of questions covered in documentation 
+in Github issues, and number of pull requests from new contributors) monthly after the documentation is published.
+
+We would consider the project successful if, after publication of the new documentation each of the following happens:
```
Comment (joestringer) [inline Documentation/contributing/development/season_of_docs.rst:69]: I presume that these success criteria are something like monthly rates at the beginning & end of the project? Do they account for existing trends?

**Review by joestringer — APPROVED** [review]
Thanks Bill!

The docs checks from github at the bottom have some false positives in them, but the documentation github action did pick up some spelling inconsistencies: https://github.com/cilium/cilium/runs/5646735171?check_suite_focus=true. You may already be aware, but some folks do not realise that one of the checks below also rendersa a preview on Netlify, so you can see how the page came out live. It's also possible to render these locally & get live updates. Let us know if you'd like any pointers on that.

One more heads up, if we merge this, it will show up at https://docs.cilium.io/en/latest for now only. If you would like to see this on https://docs.cilium.io/en/stable (or https://docs.cilium.io/en/v1.11), we will need to backport it. You can initiate this process by adding the `needs-backport/1.11` label to the PR.

I presume that you'll keep an eye out for this and once we complete the season of docs, update or remove this page?

I looked briefly through the writeup and the problem statement & how we can improve the docs makes a lot of sense to me. I don't have a lot of experience dealing with GSoC / GSoD but if you'd like to discuss that, I think @pchaigno would be a great person to talk to. I also don't know much about how realistic the specific targets or budgets are, but I imagine you (perhaps already) discussed with @lizrice .

File: Documentation/contributing/development/season_of_docs.rst (hunk 1 above)
Comment (lizrice) [inline Documentation/contributing/development/season_of_docs.rst]:
```suggestion
Cilium (current version 1.11, first released in 2017) is an Apache 2.0-licensed project for providing, 
```

File: Documentation/contributing/development/season_of_docs.rst [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+- The number of Slack questions covered in the documentation decreases by 10%
+- The number of Github issues covered in the documentation decreases by 20%
+- The number of pull requests from new contributors increases by 15%
+
+Timeline
+~~~~~~~~
+
+The project itself will take approximately six months to complete. Once the tech writer is hired, we'll spend a month on 
+tech writer orientation, then move onto the audit and inventory, and spend the last two months focusing on restructuring the documentation.
```
Comment (lizrice) [inline Documentation/contributing/development/season_of_docs.rst]:
```suggestion
tech writer orientation, then move onto the audit and inventory, and spend the last three months focusing on restructuring the documentation.
```

File: Documentation/contributing/development/season_of_docs.rst (hunk 2 above)
Comment (lizrice) [reply Documentation/contributing/development/season_of_docs.rst:69]: Yes, we'll need to make some estimate of this before the project starts (but we can do that after we've made the application to GSoD)

**Review by lizrice — APPROVED** [review]
Couple tiny changes suggested. Will be awesome to get the docs easier to navigate!

File: Documentation/contributing/development/season_of_docs.rst [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+About Our Organization
+~~~~~~~~~~~~~~~~~~~~~~
+
+Cilium (current version 1.12, first released in 2017) is an Apache 2.0-licensed project for providing, 
+securing, and observing network connectivity between container workloads - cloud native, and built around 
+the revolutionary Linux Kernel technology eBPF. Our users adopt Cilium to solve their cloud networking, 
+observability, and security issues at scale including running some of the largest Kubernetes clusters in the world. 
+Cilium has a large community of contributors from over 50 countries around the world. Multiple companies 
+have staff paid to work on Cilium too. Cilium is an incubating project in the Cloud Native Computing Foundation 
+and is the networking dataplane for Google Kubernetes Engine and Anthos.
```
Comment (xmulligan) [inline Documentation/contributing/development/season_of_docs.rst]:
```suggestion
and is the networking data plane for Google Kubernetes Engine and Anthos.
```

File: Documentation/contributing/development/season_of_docs.rst [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+About the Problem
+~~~~~~~~~~~~~~~~~
+
+Cloud native networking and security are complex topics that even long time practitioners struggle to master.
+Cilium documentation has grown organically over time and the project has mainly been used by advanced cloud native users. 
+The documentation lacks the structure that new, and even existing, users can rely on to quickly find the information they need. 
+In addition, users sometimes struggle to discover all of the features and functionality of Cilium because the documentation 
+doesn’t make them easily discoverable. Finally, new users often have questions about how to troubleshoot setting up their network.
```
Comment (xmulligan) [inline Documentation/contributing/development/season_of_docs.rst]:
```suggestion
does not make them easily discoverable. Finally, new users often have questions about how to troubleshoot setting up their network.
```

File: Documentation/contributing/development/season_of_docs.rst (hunk 1 above)
Comment (xmulligan) [reply Documentation/contributing/development/season_of_docs.rst]: thanks updated it to 1.11

**Comment by xmulligan** [comment]
@joestringer thanks for the feedback! I'm not sure if I can add the label. Do I need certain permissions to be able to do that?

It would also be great to understand where the Netlify preview is. I didn't know it did that. I would also be happy to add it to the docs after you show me so other people can find it too :)

**Comment by pchaigno** [comment]
@xmulligan The docs preview is this item in the list of GitHub checks at the bottom:
image

See also the docs build error: https://github.com/cilium/cilium/runs/5658784194?check_suite_focus=true. You're going to need to add some additional words to the dictionary with the suggested command.

For reviewers: the checkpatch failure is safe to ignore IMO. Just complaining about case on Co-authored-by.

File: Documentation/contributing/development/index.rst [hunk 6]
```diff
@@ -25,3 +25,4 @@ understand the workflow of getting these changes reviewed and merged upstream.
    debugging
    hubble
    documentation
+   season_of_docs
```
Comment (pchaigno) [inline Documentation/contributing/development/index.rst:28]: So it's currently in the Development Guide: 


Would it make more sense in the Community section above? /cc @joestringer

**Review by pchaigno — APPROVED** [review]
One suggestion on placement in the index. Content looks good to me.

**Comment by lizrice** [comment]
I was thinking it's good in proximity to the "Documentation style" item, though agree it could fit in either place. Hopefully once this docs re-structuring project happens, it will be easier to find the proper place for this page 😀

**Comment by joestringer** [comment]
Ah yes, maybe only commiters are able to set labels. Given that this PR will be linked on the 'latest' page from other websites, I think this is fine. I also briefly discussed with @pchaigno offline and it didn't seem like it would make much of a difference to also expose it on the stable docs version.

**Pull request merged**